docker/test/performance-comparison/perf.py

```diff
@@ -171,12 +173,9 @@ def substitute_parameters(query_templates, other_templates = []):
 settings = root.findall('settings/*')
 for conn_index, c in enumerate(all_connections):
     for s in settings:
-        try:
-            q = f"set {s.tag} = '{s.text}'"
-            c.execute(q)
-            print(f'set\t{conn_index}\t{c.last_query.elapsed}\t{tsv_escape(q)}')
-        except:
-            print(traceback.format_exc(), file=sys.stderr)
+        # requires clickhouse-driver >= 1.1.5 to accept arbitrary new settings
+        # (https://github.com/mymarilyn/clickhouse-driver/pull/142)
+        c.settings[s.tag] = s.text
```

Comment:
It silently ignores any unknown settings, this is bad. 
We need diagnostics in this case, and also the ability to work on one server when the other doesn't understand the setting (useful for cases when you're adding a new one).

It would be ideal if we could make it not to reconnect in case of errors, but I didn't find a way to do this.

Reply:
Accidentally I also tried to commit a fix for this to master and this is why I noticed this PR :)

My fix just re-applied the settings before each query, maybe I'll use it instead.

Reply:
>My fix just re-applied the settings before each query, maybe I'll use it instead.

But this still does not guarantee that the setting will be applied

>It silently ignores any unknown settings, this is bad.

Maybe the server should have some config option to disallow unknown settings? This looks more elegant to me.

>It would be ideal if we could make it not to reconnect in case of errors, but I didn't find a way to do this.

And/or we can add such support to clickhouse-driver, looks pretty easy

Reply:
Apparently there is support for erroring out on unknown settings in the protocol -- the client must mark them as "important" (I'm looking at `BaseSettings<Traits_>::read(ReadBuffer & in, SettingsWriteFormat format)`).

Reply:
>Apparently there is support for erroring out on unknown settings in the protocol -- the client must mark them as "important" (I'm looking at BaseSettings<Traits_>::read(ReadBuffer & in, SettingsWriteFormat format)).

Indeed, then the driver can be updated, looks pretty trivial - https://github.com/mymarilyn/clickhouse-driver/compare/master...azat-archive:settings-is-important

*P.S. patch is completely untested, and requires some trivial tests*

Reply:
>Indeed, then the driver can be updated, looks pretty trivial - mymarilyn/clickhouse-driver@master...azat-archive:settings-is-important

Here is a patch - https://github.com/mymarilyn/clickhouse-driver/pull/173